### PR TITLE
[CSL 538] Add Support for Hidden Fields

### PR DESCRIPTION
**Pull request by jjl014**
#### Updates:
* Add support for hidden fields to Search/Browse/Autocomplete requests
* Add tests (all tests passing)

### CODE REVIEW

File: AutocompleteClient/FW/Logic/Request/CIOSearchQuery.swift [hunk 1]
```diff
@@ -56,23 +61,25 @@ public struct CIOSearchQuery: CIORequestData {
...
+        - hiddenFields: The list of hidden metadata fields to return
      
      ### Usge Example: ###
      ```
      let facetFilters = [(key: "Nutrition", value: "Organic"),
                          (key: "Nutrition", value: "Natural"),
                          (key: "Brand", value: "Kraft Foods")]
 
-     let searchQuery = CIOSearchQuery(query: "red", filters: CIOQueryFilters(groupFilter: nil, facetFilters: facetFilters), page: 1, perPage: 30, section: "Products")
+     let searchQuery = CIOSearchQuery(query: "red", filters: CIOQueryFilters(groupFilter: nil, facetFilters: facetFilters), page: 1, perPage: 30, section: "Products", hiddenFields: ["hiddenField1", "hiddenField2"])
```
Comment (rubikzube) [inline AutocompleteClient/FW/Logic/Request/CIOSearchQuery.swift:72]: Let's get a more descriptive example

File: AutocompleteClient/FW/Logic/Request/CIOBrowseQuery.swift [hunk 2]
```diff
@@ -62,24 +67,26 @@ public struct CIOBrowseQuery: CIORequestData {
...
+        - hiddenFields: The list of hidden metadata fields to return
 
      ### Usage Example: ###
      ```
      let facetFilters = [(key: "Nutrition", value: "Organic"),
                          (key: "Nutrition", value: "Natural"),
                          (key: "Brand", value: "Kraft Foods")]
 
-     let browseQuery = CIOBrowseQuery(filterName: "group_id", filterValue: "Pantry", filters: CIOQueryFilters(groupFilter: nil, facetFilters: facetFilters), page: 1, perPage: 30, section: "Products")
+     let browseQuery = CIOBrowseQuery(filterName: "group_id", filterValue: "Pantry", filters: CIOQueryFilters(groupFilter: nil, facetFilters: facetFilters), page: 1, perPage: 30, section: "Products", hiddenFields: ["hiddenField1", "hiddenField2"])
```
Comment (rubikzube) [inline AutocompleteClient/FW/Logic/Request/CIOBrowseQuery.swift:78]: Let's get a more descriptive example

File: AutocompleteClient/FW/Logic/Request/CIOAutocompleteQuery.swift [hunk 3]
```diff
@@ -44,23 +49,26 @@ public struct CIOAutocompleteQuery: CIORequestData {
         - numResults: The number of results to return
         - numresultsForSection: The section to return results from
         - filters: The filters used to refine results
+        - hiddenFields: The list of hidden metadata fields to return
      
      ### Usage Example: ###
      ```
-     let autocompleteQuery = CIOAutocompleteQuery(query: "apple", numResults: 5, numResultsForSection: ["Products": 6, "Search Suggestions": 8])
+     let autocompleteQuery = CIOAutocompleteQuery(query: "apple", numResults: 5, numResultsForSection: ["Products": 6, "Search Suggestions": 8], hiddenFields: ["hiddenField1", "hiddenField2"])
```
Comment (rubikzube) [inline AutocompleteClient/FW/Logic/Request/CIOAutocompleteQuery.swift:56]: Let's get a more descriptive example

File: AutocompleteClientTests/FW/Logic/Request/AutocompleteQueryRequestBuilderTests.swift [hunk 4]
```diff
@@ -130,4 +130,18 @@ class AutocompleteQueryRequestBuilderTests: XCTestCase {
         XCTAssertTrue(url.contains("key=\(testACKey)"), "URL should contain api key.")
         XCTAssertEqual(request.httpMethod, "GET")
     }
+
+    func testAutocompleteQueryBuilder_WithHiddenFields() {
+        let hiddenFields = ["hiddenField1", "hiddenField2"]
```
Comment (rubikzube) [inline AutocompleteClientTests/FW/Logic/Request/AutocompleteQueryRequestBuilderTests.swift:135]: Let's get a more descriptive example

**Review by rubikzube — CHANGES_REQUESTED** [review]

**Comment by jjl014** [comment]
Update: Hidden fields is not supported on the recommendations endpoint.

**Review by rubikzube — APPROVED** [review]

**Review by rubikzube — APPROVED** [review]

**Pull request merged**